Extraction failing due to runtime error

I noted today that extractions seem to be failing with the following error. I am using the hotio nightly version of the unpackerr container image.

```
[INFO] 2024/07/18 05:45:14 [Unpackerr] Queue: 1 waiting, 0 queued, 0 extracting, 0 extracted, 0 imported, 0 failed, 0 deleted
[INFO] 2024/07/18 05:46:14 [Sonarr] Extraction Queued: /data/torrents/tv/show-ep, retries: 0, 1 archive: /data/torrents/tv/show-ep.rar, delete orig: false, queue size: 1
panic: runtime error: index out of range [-1]

goroutine 21 [running]:
github.com/Unpackerr/unpackerr/pkg/unpackerr.(*Unpackerr).updateHistory(...)
        /unpackerr/pkg/unpackerr/tray_other.go:30
github.com/Unpackerr/unpackerr/pkg/unpackerr.(*Unpackerr).logQueuedDownload(0xc000128d90, 0x1, 0xc000218200, 0xc0001beb10)
        /unpackerr/pkg/unpackerr/handlers.go:149 +0x475
github.com/Unpackerr/unpackerr/pkg/unpackerr.(*Unpackerr).extractCompletedDownload(0xc000128d90, {0xc000208150, 0x2a}, {0x5?, 0x7?, 0x11837a0?}, 0xc000218200)
        /unpackerr/pkg/unpackerr/handlers.go:138 +0x49d
github.com/Unpackerr/unpackerr/pkg/unpackerr.(*Unpackerr).extractCompletedDownloads(0xc000128d90, {0xc0000b1e54?, 0x0?, 0x11837a0?})
        /unpackerr/pkg/unpackerr/handlers.go:86 +0xd6
github.com/Unpackerr/unpackerr/pkg/unpackerr.(*Unpackerr).Run(0xc000128d90)
        /unpackerr/pkg/unpackerr/start.go:295 +0x3fd
created by github.com/Unpackerr/unpackerr/pkg/unpackerr.(*Unpackerr).startTray in goroutine 1
        /unpackerr/pkg/unpackerr/tray_other.go:12 +0x73
[INFO] 2024/07/18 05:46:15 Unpackerr v58e050870d5b00fdcd52c99c482ea3208dee9004- Starting! PID: 171, UID: 13009, GID: 100, Umask: 2, Now: 2024-07-18 05:46:16 -0600 MDT
```

--- a/pkg/unpackerr/history.go
+++ b/pkg/unpackerr/history.go
@@ -1,0 +1,48 @@
+package unpackerr
+
+import (
+	"strconv"
+
+	"github.com/Unpackerr/unpackerr/pkg/ui"
+)
+
+// Safety constants.
+const (
+	hist     = "hist_"
+	histNone = "hist_none"
+)
+
+// History holds the history of extracted items.
+type History struct {
+	Items    []string
+	Finished uint
+	Retries  uint
+	Map      map[string]*Extract
+}
+
+// This is called every time an item is queued.
+func (u *Unpackerr) updateHistory(item string) {
+	if u.KeepHistory == 0 {
+		return
+	}
+
+	if ui.HasGUI() && item != "" {
+		u.menu[histNone].Hide()
+	}
+
+	u.History.Items[0] = item
+
+	//nolint:intrange // Do not process 0; this isn't an `intrange`.
+	for idx := len(u.History.Items) - 1; idx > 0; idx-- {
+		// u.History.Items is a slice with a set (identical) length and capacity.
+		switch u.History.Items[idx] = u.History.Items[idx-1]; {
+		case !ui.HasGUI():
+			continue
+		case u.History.Items[idx] != "":
+			u.menu[hist+starr.Itoa(idx)].SetTitle(u.History.Items[idx])
+			u.menu[hist+strconv.Itoa(idx)].Show()
+		default:
+			u.menu[hist+strconv.Itoa(idx)].Hide()
+		}
+	}
+}

--- a/pkg/unpackerr/history.go
+++ b/pkg/unpackerr/history.go
@@ -32,7 +32,7 @@ func (u *Unpackerr) updateHistory(item string) {
 
 	u.History.Items[0] = item
 
-	//nolint:intrange // Do not process 0; this isn't an `intrange`.
+	// Do not process 0; this isn't an `intrange`.
 	for idx := len(u.History.Items) - 1; idx > 0; idx-- {
 		// u.History.Items is a slice with a set (identical) length and capacity.
 		switch u.History.Items[idx] = u.History.Items[idx-1]; {

--- a/pkg/unpackerr/history.go
+++ b/pkg/unpackerr/history.go
@@ -39,7 +39,7 @@ func (u *Unpackerr) updateHistory(item string) {
 		case !ui.HasGUI():
 			continue
 		case u.History.Items[idx] != "":
-			u.menu[hist+starr.Itoa(idx)].SetTitle(u.History.Items[idx])
+			u.menu[hist+strconv.Itoa(idx)].SetTitle(u.History.Items[idx])
 			u.menu[hist+strconv.Itoa(idx)].Show()
 		default:
 			u.menu[hist+strconv.Itoa(idx)].Hide()

--- a/pkg/unpackerr/start.go
+++ b/pkg/unpackerr/start.go
@@ -85,14 +85,6 @@ type Flags struct {
 	webhook    uint
 }
 
-// History holds the history of extracted items.
-type History struct {
-	Items    []string
-	Finished uint
-	Retries  uint
-	Map      map[string]*Extract
-}
-
 // New returns an UnpackerPoller struct full of defaults.
 // An empty struct will surely cause you pain, so use this!
 func New() *Unpackerr {

--- a/pkg/unpackerr/tray.go
+++ b/pkg/unpackerr/tray.go
@@ -18,12 +18,6 @@ import (
 	"golift.io/version"
 )
 
-// Safety constants.
-const (
-	hist     = "hist_"
-	histNone = "hist_none"
-)
-
 // startTray Run()s readyTray to bring up the web server and the GUI app.
 func (u *Unpackerr) startTray() {
 	if !ui.HasGUI() {
@@ -252,36 +246,5 @@ func (u *Unpackerr) checkForUpdate() {
 		version.Version, version.Revision, update.Current, update.RelDate.Format("Jan 2, 2006"), ago)
 	if yes {
 		_ = ui.OpenURL(update.CurrURL)
-	}
-}
-
-// This is called every time an item is queued.
-func (u *Unpackerr) updateHistory(item string) {
-	if u.KeepHistory == 0 {
-		return
-	}
-
-	if ui.HasGUI() && item != "" {
-		u.menu[histNone].Hide()
-	}
-
-	// u.History.Items is a slice with a set (identical) length and capacity.
-	for idx := range u.History.Items {
-		if idx == 0 {
-			u.History.Items[0] = item
-		} else {
-			u.History.Items[idx] = u.History.Items[idx-1]
-		}
-
-		if !ui.HasGUI() {
-			continue
-		}
-
-		if u.History.Items[idx] != "" {
-			u.menu[hist+strconv.Itoa(idx)].SetTitle(u.History.Items[idx])
-			u.menu[hist+strconv.Itoa(idx)].Show()
-		} else {
-			u.menu[hist+strconv.Itoa(idx)].Hide()
-		}
 	}
 }

--- a/pkg/unpackerr/tray_other.go
+++ b/pkg/unpackerr/tray_other.go
@@ -18,15 +18,3 @@ func (u *Unpackerr) startTray() {
 func (u *Unpackerr) updateTray(_ *Stats, _ uint) {
 	// there is no tray.
 }
-
-func (u *Unpackerr) updateHistory(item string) {
-	if u.KeepHistory == 0 {
-		return
-	}
-
-	u.History.Items[0] = item
-	// u.History.Items is a slice with a set (identical) length and capacity.
-	for idx := range u.History.Items {
-		u.History.Items[idx] = u.History.Items[idx-1]
-	}
-}

